threshsign/src/ThresholdSignaturesTypes.cpp

```diff
@@ -59,8 +62,8 @@ void Cryptosystem::generateNewPseudorandomKeys() {
 
   std::tie(signers, verifier)
     = factory->newRandomSigners(threshold, numSigners);
-
-  publicKey = verifier->getPublicKey().toString();
+  if (forceMultisig_ || type == THRESHOLD_BLS_SCHEME)
```

Comment:
Yes, in case of multisig, when numsigners!=threshold public key is dynamically calculated and getPublicKey() for the entire system doesn't make sense.

Reply:
Thank you for the clarification.
I will mention the current handling of the optional-ness of the public key does not seem exceedingly clean to me, but if you are planning to switch to only using multisig soon (at which point we presumably won't store a public key in any case), I might refrain from commenting on this until the transition is complete given we could possibly remove the public key and its handling entirely at some point.

Reply:
Great